### PR TITLE
Test new dbms

### PR DESCRIPTION
**Pull request by rory-hr**
This merge will add a number of changes to the repository:
1. Rebuild the server including all API routes
2. Remove MySQL and replace with MongoDB
3. Add a MongoDB database seeding script
4. Refactor the client to accept new form of data according to new API
5. Apply style corrections according to AirBnB style guide

### CODE REVIEW

File: client/dist/index.html [hunk 1]
```diff
@@ -11,6 +11,7 @@
   <div id="app"></div>
   <div id="modal-root"></div>
   <script src="bundle.js"></script>
+  <!-- Icons made by <a href="https://www.flaticon.com/authors/freepik" title="Freepik">Freepik</a> from <a href="https://www.flaticon.com/" title="Flaticon"> www.flaticon.com</a> -->
```
Comment (lheddleston) [inline client/dist/index.html:14]: Did you mean to include this as a comment?

Comment (rory-hr) [reply client/dist/index.html:14]: Hi Laura, yes. I'm using these icons and wanted to add attribution when I get around to it. Good catch.

File: database/dbseeding/fakerAlteration.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  if (typeof format !== "number") {
+    format = faker.random.number(formats.length - 1);
+  }
+
+  return f(formats[format]);
+}
+
+const pathToWords = 'faker/lib/locales/en/lorem/words.js'
+const words = [  "delicious",
```
Comment (lheddleston) [inline database/dbseeding/fakerAlteration.js:29]: I love this! 🌯

Comment (rory-hr) [reply database/dbseeding/fakerAlteration.js:29]: I had some fun with it but my creativity eventually wore off lol

File: database/dbseeding/mongoModels.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  photoUrl: { type: String, required: true },
+  caption: String,
+  createdAt: Number,
+  updatedAt: Number,
+  rating: { type: Number, required: true },
+  reviewText: String,
+});
+
+// const photosSchema = new Schema({
```
Comment (lheddleston) [inline database/dbseeding/mongoModels.js]: Do you need this anymore? You've included your photo data in your reviewsSchema, right?

Comment (rory-hr) [reply database/dbseeding/mongoModels.js]: I will flag this for removal. Thanks

File: server/controllers/mongo.controller.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  const deletedReview = req.body;
+  Model.deleteReview(deletedReview, (err, result) => {
+    if (err) {
+      console.log(err);
+      res.status(400).send();
+    } else {
+      res.status(200).send(result);
+    }
+  });
+};
```
Comment (lheddleston) [inline server/controllers/mongo.controller.js:123]: Clean and easy to read 👍

File: server/index.js [hunk 5]
```diff
@@ -1,76 +1,79 @@
...
+
+/*
+  * PUT ROUTES
+*/
+
+app.put('/api/dishes/', (req, res) => {
+  Controller.putDish(req, res);
+});
+
+app.put('/api/reviews/', (req, res) => {
```
Comment (lheddleston) [inline server/index.js]: The PUT routes end with '/' but the others don't. Is that a naming convention?

Comment (rory-hr) [reply server/index.js]: Thanks I fixed that and remove the commented route at the bottom.

File: server/models/mongo.model.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+      User.findByIdAndUpdate(result.userId, { $pull: { reviews: result._id } }, (err, result) => {
+        if (err) {
+          callback(err);
+        } else {
+          callback(null, result);
+        }
+      });
+    });
+  });
+};
```
Comment (lheddleston) [inline server/models/mongo.model.js:141]: 👍

**Review by lheddleston — APPROVED** [review]

File: server/index.js (hunk 5 above)
Comment (lheddleston) [reply server/index.js]: 👍

**Pull request merged**